cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -399,6 +400,12 @@ func ValidateIPNetFromString(subnetStr string, minAddrs int64, isDualStack bool,
 		if numAddresses < minAddrs {
 			allErrs = append(allErrs, field.Invalid(fldPath, s.String(), fmt.Sprintf("subnet with %d address(es) is too small, the minimum is %d", numAddresses, minAddrs)))
 		}
+
+		// Warn when the subnet is in site-local range - i.e. contains addresses that belong to FEC0::/10
+		_, siteLocalNet, _ := net.ParseCIDR("FEC0::/10")
+		if siteLocalNet.Contains(s.IP) || s.Contains(siteLocalNet.IP) {
+			klog.Warningf("the subnet %v contains IPv6 site-local addresses that belong to FEC0::/10 which is deprecated", s)
```

Comment:
Some comments on the warning message and is lgtm

lowercase is the recommendation for ipv6 addesses representation we should follow it
```
4.3.  Lowercase

   The characters "a", "b", "c", "d", "e", and "f" in an IPv6 address
   MUST be represented in lowercase.
```
https://tools.ietf.org/html/rfc5952

Also I think that it will be good to add a reference to the rfc3879, the one that deprecated it, to avoid confusion, I can see how some people will think that it was a kubernetes decision :smile:

Reply:
i was going to suggest lowercase too.

Reply:
Done